scipy/stats/tests/test_stats.py

```diff
@@ -1186,17 +1187,40 @@ def test_weightedtau():
 
 
 def test_kendall_tau_large():
-    n = 172.
-    x = np.arange(n)
-    y = np.arange(n)
-    _, pval = stats.kendalltau(x, y, method='exact')
-    assert_equal(pval, 0.0)
-    y[-1], y[-2] = y[-2], y[-1]
-    _, pval = stats.kendalltau(x, y, method='exact')
-    assert_equal(pval, 0.0)
-    y[-3], y[-4] = y[-4], y[-3]
-    _, pval = stats.kendalltau(x, y, method='exact')
-    assert_equal(pval, 0.0)
+    # n = 172.
+    # x = np.arange(n)
+    # y = np.arange(n)
+    # _, pval = stats.kendalltau(x, y, method='exact')
+    # assert_equal(pval, 0.0)
+    # y[-1], y[-2] = y[-2], y[-1]
+    # _, pval = stats.kendalltau(x, y, method='exact')
+    # assert_equal(pval, 0.0)
+    # y[-3], y[-4] = y[-4], y[-3]
+    # _, pval = stats.kendalltau(x, y, method='exact')
+    # assert_equal(pval, 0.0)
+
+    # Test for the exact method with large samples (n > 171)
+    # expected values generated using SymPy
+    expectations = {100: (-0.033131313131313136, 0.62822615287956040664),
+                    101: (0.03524752475247526, 0.60439525773513602669),
+                    200: (-0.015376884422110552, 0.74753983745929675209),
+                    201: (-0.03920398009950248, 0.40959218958120363618),
+                    400: (0.02343358395989975, 0.48444283672113314099),
+                    401: (-0.014563591022443893, 0.66363159823474837662),
+                    800: (-0.018948685857321654, 0.42265448483120932055),
+                    801: (-0.014675405742821473, 0.53437553412194416236)}
```

Comment:
Are these the same values as in `test_mstats_basic.py`? If so, can you combine in any way to avoid the duplication?

Reply:
One way to combine would be to factor into 3 tests: test the low-level `_kendall_p_exact`, and then tests of `stats.kendalltau` and `mstats.kendalltau` (to test the masking), which no longer would need to be as extensive.

Reply:
Sounds good, spilt in the latest commit, so the duplication is now avoided. The input to the new test looks a bit different since it calls the p-value calculation directly, but corresponds to the cases used before.